components/ModSubmissionListContainer.vue

```diff
@@ -141,17 +143,27 @@ const submissionListItemTableColumns = computed(() => {
 })
 
 const handleClickToSubmissionForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditSubmission)
     modSubmissionsListStore.selectedSubmissionId = id
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditSubmission)
 }
 
 const handleClickToHealthcareProfessionalForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditHealthcareProfessional)
-    modSubmissionsListStore.selectedHealthcareProfessionalId = id
+    healthcareProfessionalsStore.selectedHealthcareProfessionalId = id
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditHealthcareProfessional)
 }
 
 const handleClickToFacilityForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditFacility)
-    modSubmissionsListStore.selectedFacilityId = id
+    facilitiesStore.selectedFacilityId = id
+    facilitiesStore.setSelectedFacilityData(id)
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditFacility)
 }
+
+/** This ensures that the data for the selected facility is loaded before mounting the component,
```

Comment:
Non-blocking: I don't think it has been formally decided if comments in our `.vue` files should be written with `//` or `/** **/`, but we should probably pick one and stick with it throughout the codebase. Something we should probably discuss on Slack!